workspaces/ui-v2/src/pages/docs/components/ShapeEditor.tsx

```diff
@@ -149,32 +231,21 @@ const FieldEditor: FC<{
   return (
     <div className={classes.editor}>
       <ButtonGroup className={classes.typeSelector}>
-        <Button
-          disableElevation
-          variant={
-            currentJsonTypes.includes(JsonType.UNDEFINED)
-              ? 'contained'
-              : 'outlined'
-          }
-          startIcon={
-            currentJsonTypes.includes(JsonType.UNDEFINED) ? <CheckIcon /> : null
-          }
-          onClick={onClickTypeButton(JsonType.UNDEFINED)}
-        >
-          Optional
-        </Button>
-        <Button
-          disableElevation
-          variant={
-            currentJsonTypes.includes(JsonType.NULL) ? 'contained' : 'outlined'
-          }
-          startIcon={
-            currentJsonTypes.includes(JsonType.NULL) ? <CheckIcon /> : null
-          }
-          onClick={onClickTypeButton(JsonType.NULL)}
-        >
-          Null
-        </Button>
+        {[...editableTypes].map((editableType) => (
+          <Button
+            key={editableType}
+            disableElevation
+            variant={
+              currentJsonTypes.includes(editableType) ? 'contained' : 'outlined'
+            }
+            startIcon={
+              currentJsonTypes.includes(editableType) ? <CheckIcon /> : null
+            }
+            onClick={onClickTypeButton(editableType)}
+          >
+            {editableType === JsonType.UNDEFINED ? 'Optional' : editableType}
+          </Button>
+        ))}
```

Comment:
I intentionally kept this code wet, as the UI for this is a place holder, and being explicit is well worth it over being dry, especially this far as the edge of the UI. So you might see this undone in the near future 😅.

Reply:
That's completely fine - I think the reason I update this is because there's other variables to make this work (i.e. filtering our specific keys) - so moving these to a single place.

I agree though, some places it's better to be explicit, and this might be one of them.